servers/server.go

```diff
@@ -601,6 +611,138 @@ func (p *Server) Extract(source, destination string) error {
 	return files.Extract(p.GetFileServer(), source, destination, "*", false, nil)
 }
 
+func (p *Server) CreateBackup() (string, int64, error) {
+	sourceFiles := []string{p.GetFileServer().Prefix()}
+
+	backupDirectory := p.RunningEnvironment.GetBackupDirectory()
+	if backupDirectory == "" {
+		return "", 0, pufferpanel.ErrSettingNotConfigured("backupDirectory")
+	}
+
+	if p.GetEnvironment().IsBackingUp() {
+		return "", 0, pufferpanel.ErrBackupInProgress
+	}
+
+	p.GetEnvironment().SetBackingUp(true)
+	defer p.GetEnvironment().SetBackingUp(false)
+
+	_, err := os.Stat(backupDirectory)
+	if err != nil && os.IsNotExist(err) {
+		err = os.Mkdir(backupDirectory, 0755)
+		if err != nil && !os.IsExist(err) {
+			return "", 0, err
+		}
+	}
+
+	backupId, err := uuid.NewV4()
+	if err != nil {
+		return "", 0, err
+	}
+	backupFileName := backupId.String() + ".tar.sz"
```

Comment:
tar.sz?

Reply:
its S2 snappy compression, I had a look at the compress package for the best speed vs compression. 
did this as a last change as didn't know if performance for backing up and restoring vs compatibility was best.
Can move it back to tar.gz if that is preferred?

Reply:
Fair enough, didn't see sz before. I think speed is probably better in this case.